ledger/participant-integration-api/src/main/resources/db/migration/h2database-appendonly/V1__Append_only_schema.sql

```diff
@@ -106,11 +106,25 @@ CREATE TABLE participant_command_submissions (
 CREATE TABLE participant_command_completions (
     completion_offset VARCHAR NOT NULL,
     record_time TIMESTAMP NOT NULL,
-    application_id VARCHAR NOT NULL,
+    -- The application ID has to be provided by the application.
```

Comment:
discussions on the precision made me think more about this:
* Foreword: sorry to raise this so late, but this is exactly what is my main concern regarding too granular PRs: not knowing the whole picture makes it quite hard to reason about this
* It seems to me that all the old fields are pass-through info for the indexer (except from the offset). How about the new ones? Does the indexer need to be aware of all this info? Can't we just squeeze all the non-relevant fields into a proto which could be possibly passed through the system? (this pattern can be captured in all the contract related serialized data and also in here since @SamirTalwar-DA 's commit, the rejection_status_details)

Reply:
My general idea is that everything that needs to be serialized (e.g. b/c it's driver-dependent) should be, but nothing more because that makes using the info more complicated.

In this case, it's the new command dedup model and it's API behavior, so I think we shouldn't make it opaque at the index level and it's supposed not to change in the (near) future; for this reason, maintenance should also be minimal after this initial work.

Being core info, it's also possible that the indexer may benefit from being aware of these new fields in the future (although the design trend for command deduplication implementations is currently driver-based). In order to make it easy to use (and to migrate) them in future designs/architectures, I think it's better if they are known to the DB.
Another use case is debugging/troubleshooting; it is true that there are other means (e.g. logging) but being able to look at the actual data in the DB at a given point in time can be pretty useful in some circumstances.

I am aware that even just adding pass-through fields is a lot of work and code but maybe that's the actual issue and we should rather deal with that.

Does it make sense?

Reply:
After discussion with @meiersi-da and @rautenrieth-da the concolusion is that it is okay in this case to keep this pass-thru data flattened :ok_hand: